[CORL-794] Create a common date formatter

nick-funk opened the pull request:
## What does this PR do?

Updates all uses of Intl.DateTimeFormat(...)
with re-usable date time formatter functions.

The formatter functions have default formatting
options, but these can be overridden.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

Look around the site at various dates (comment stream time stamps is a good area) and see that the new consolidated date formatting is equivalent to before.

nick-funk committed:
Create a common date formatter

Updates all uses of Intl.DateTimeFormat(...)
with re-usable date time formatter functions.

The formatter functions have default formatting
options, but these can be overridden.

CORL-794

wyattjoh commented on src/core/client/ui/hooks/useDateTimeFormat.ts:
```diff
@@ -1,11 +1,9 @@
-    locales,
-    options,
-  ]);
+  return useMemo(() => createDateFormatter(locales, options), [locales]);
```
```suggestion
  return useMemo(() => createDateFormatter(locales, options), [locales, options]);
```

`options` must be included as a dependency in `useMemo` in case the value ever changes.

wyattjoh commented on src/core/common/date.ts:
```diff
@@ -0,0 +1,20 @@
+export function createDateFormatter(
+  locales: string[] | undefined,
```
suggestion: because technically `Intl.DateTimeFormat` can accept `string | string[] | undefined`, modifying this type will allow the other uses of it (like in `src/core/server/services/users/download/download.ts`) to just pass the `string` instead.

Same thing with the `formatDate` function here (that's used in `src/core/server/services/users/users.ts`).

nick-funk committed:
Include options in useMemo

Co-authored-by: Wyatt Johnson <wyattjoh@gmail.com>

nick-funk committed:
Make date formatting accept single locale string

Makes formatDate and createDateFormatter accept
string | string[] | undefined for the locales
argument.

CORL-794

kodiakhq[bot] committed:
Merge branch 'master' into bug/CORL-794-date-format
